### PR TITLE
deprecate-nginx-id-from-violation

### PR DESCRIPTION
**Pull request by mohamed-gougam**
### Proposed changes

We previously confirmed that it’s impossible to gather the NGINX ID for the NAP Violation being generated and sent to nms.

This follow up ticket is to deprecate the NGINX ID from both the violation struct being transferred.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

File: sdk/proto/events/event.proto [hunk 1]
```diff
@@ -115,7 +115,7 @@ message SecurityViolationEvent {
   string InstanceTags = 36 [(gogoproto.jsontag) = "instance_tags"];
   string InstanceGroup = 37 [(gogoproto.jsontag) = "instance_group"];
   string DisplayName = 38 [(gogoproto.jsontag) = "display_name"];
-  string NginxID = 39 [(gogoproto.jsontag) = "nginx_id"];
+  string NginxID = 39 [(gogoproto.jsontag) = "nginx_id", deprecated=true];
```
[the remark this answers is not in the document]
Comment (mohamed-gougam) [reply sdk/proto/events/event.proto]: applied 👍

**Review by olli-holmala — APPROVED** [review]

File: sdk/proto/events/event.proto [hunk 2]
```diff
@@ -66,6 +66,10 @@ message ActivityEvent {
 
 // Represents a security violation that is emitted by the agent
 message SecurityViolationEvent {
+  // DEPRECATED
```
Comment (sanathkumarbs) [inline sdk/proto/events/event.proto:69]: question: was the protobuf bindings regenerated and the vendor too after the change from `deprecated` to the `reserved` field approach that you took?

File: sdk/proto/events/event.proto [hunk 3]
```diff
@@ -66,6 +66,10 @@ message ActivityEvent {
...
+  reserved 39;
+  reserved "NginxID";
```
Comment (sanathkumarbs) [inline sdk/proto/events/event.proto:71]: question to @oliveromahony / @dhurley and the rest of the agent team - how do we add an RN note for this?

**Review by sanathkumarbs — APPROVED** [review]
LGTM, thank you for doing this!

File: sdk/proto/events/event.proto (hunk 2 above)
Comment (mohamed-gougam) [reply sdk/proto/events/event.proto:69]: Yes, when used `deprecated` that reflected only documentation wise. When used `reserved` the field is not in the message anymore, can't be used or set. So yep the generated struct for `SecurityViolationEvent` did change.

File: sdk/proto/events/event.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanathkumarbs) [reply sdk/proto/events/event.proto]: +1 to @dhurley's comment

File: sdk/proto/events/event.proto (hunk 2 above)
Comment (sanathkumarbs) [reply sdk/proto/events/event.proto:69]: Thanks, @mohamed-gougam

File: sdk/proto/events/event.proto (hunk 3 above)
Comment (sanathkumarbs) [reply sdk/proto/events/event.proto:71]: @mohamed-gougam, no need to wait on this to be resolved for the PR. You can follow up with a documentation update or an RN update. I'll resolve this thread for now, but let's figure this out as a follow on.

**Review by oliveromahony — APPROVED** [review]

**Review by dhurley — APPROVED** [review]

**Pull request merged**